src/compress/zip/checksum_reader.cr

```diff
@@ -5,14 +5,14 @@ module Compress::Zip
   private class ChecksumReader < IO
     getter crc32 = ::Digest::CRC32.initial
 
-    def initialize(@io : IO, @filename : String, verify @expected_crc32 : UInt32? = nil)
+    def initialize(@io : IO, @filename : Path | String, verify @expected_crc32 : UInt32? = nil)
```

Comment:
I don't think there's much value in preserving the type union into the instance variable. Let's convene on a single type here (probably `String`?).